src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -996,7 +1010,10 @@ protected override void ProcessRecord()
 
                     try
                     {
-                        result = SessionState.Path.SetLocation (relationshipPath, CmdletProviderContext);
+                        relationshipPath = ReplacePathWithLastLocationIfApplicable(relationshipPath);
+                        var initialPath = SessionState.Path.CurrentLocation;
+                        result = SessionState.Path.SetLocation(relationshipPath, CmdletProviderContext);
```

Comment:
The previous comments apply here, including moving the logic to `SessionState.Path.SetLocation`. In fact, I think if you did that, the change here wouldn't be necessary.